runtime/src/main/kotlin/kotlin/native/concurrent/Freezing.kt

```diff
@@ -42,12 +42,25 @@ public fun <T> T.freeze(): T {
 public val Any?.isFrozen
     get() = isFrozenInternal(this)
 
+
+/**
+ * Transforms an object to the frozen form by creating frozen version of the object, by either
+ * deep copy of mutable objects and reusing frozen references. Note, that consequent calls
+ * of this function may return different instances.
+ *
+ * @return the frozen object form
+ * @see ensureNeverFrozen
+ */
+public inline fun <reified T> T.toFrozen(): T =
```

Comment:
Maybe add a note to `freeze` documentation about its side effects with reference to `toFrozen` as a safer alternative?